lib/data/entity.js

```diff
@@ -31,7 +36,29 @@ const getEntity = (entityFields, xml) => new Promise((resolve, reject) => {
     else if (field.propertyName != null)
       entity.data[field.propertyName] = text;
   });
-  stream.on('end', () => resolve(entity));
+  stream.on('end', () => {
+    // After collecting the entity data from the submission and form fields,
+    // we validate the data (properties must include dataset, label, and uuid)
+    // and assemble a valid entity data object. System properties like dataset
+    // label, uuid are in entity.system while the contents from the bound form
+    // fields are in enttiy.data.
+    const { dataset, label, id } = entity.system;
```

Comment:
can we move following lines to separate function `validatedEntity(entity)`? it will make our unit tests simpler